### ISSUE TITLE
Cannot delete tag retention rule

### ISSUE THREAD

**Issue by quentinalbertone** [goharbor/harbor#12236]
If you are reporting a problem, please make sure the following information are provided:

Hello everyone

**Expected behavior and actual behavior:**
On the web ui, I would like to delete a tag retention rule but I have a `{"code":500,"message":"Internal Server Error"}`
I was looking for a DELETE method on the swagger but I cannot find one.

But I can update each fields of the rule, for exemple disable it, but I cannot delete it.
 
**Steps to reproduce the problem:**
Delete a tag retention rule

**Versions:**
Please specify the versions of following systems.

- harbor version: goharbor/harbor-core:v1.10.1

**logs:**
this is my retention tag
```JSON
{
  "id": 1,
  "algorithm": "or",
  "rules": [
    {
      "id": 0,
      "priority": 0,
      "disabled": true,
      "action": "retain",
      "template": "always",
      "params": {
        "latestPushedK": 10
      },
      "tag_selectors": [
        {
          "kind": "doublestar",
          "decoration": "matches",
          "pattern": "**"
        }
      ],
      "scope_selectors": {
        "repository": [
          {
            "kind": "doublestar",
            "decoration": "repoMatches",
            "pattern": "**"
          }
        ]
      }
    }
  ],
  "trigger": {
    "kind": "Schedule",
    "settings": {
      "cron": "0 0 0 * * 0"
    },
    "references": {
      "job_id": 1
    }
  },
  "scope": {
    "level": "project",
    "ref": 4
  }
}
```
and the error from harbor core
```Bash
2020-06-15T09:39:19Z [ERROR] [/common/api/base.go:233]: status behind error
2020-06-15T09:39:19Z [ERROR] [/common/api/base.go:73]: PUT /api/retentions/1 failed with error: {"code":500,"message":"Internal Server Error"}
2020/06/15 09:39:19.757 [D] [server.go:2774]  |  91.69.123.122| 500 |   7.466823ms|   match| PUT      /api/retentions/1   r:/api/retentions/:id
```

Have a good day

**Comment by reasonerjt** [comment]
@quentinalbertone 

@steven-zou could you clarify if this is a known issue in v1.10.1?

**Comment by quentinalbertone** [comment]
Do you need more information ? :arrow_up:

**Comment by steven-zou** [comment]
@bitsf 

Please help to confirm if we have fixed the issue mentioned here. If not, check if we can fix it in the upcoming patches. Thanks!

**Comment by bitsf** [comment]
The simple way, you can delete it with Chrome, and see how request is send in developer tools

**Comment by agill17** [comment]
I am also running into this issue. I am trying to delete it from the UI and I keep getting "Internal server error"

**Comment by bitsf** [comment]
`status behind error` should be fixed in 2.1.2+, it is because jobservice not handle status reporting order correctly
